### PR TITLE
Upgrade webpack to v5

### PR DESCRIPTION
**Pull request by hsubox76**
See internal doc ([go/update-webpack-to-v5-firebase-js-sdk](https://goto.google.com/update-webpack-to-v5-firebase-js-sdk)) for issues upgrading so far. This is needed to allow us to upgrade to Node 18+.

Continues https://github.com/firebase/firebase-js-sdk/pull/7374

### CODE REVIEW

**Review by dwyfrequency — APPROVED** [review]
LGTM

File: packages/auth/test/helpers/integration/helpers.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 import * as sinon from 'sinon';
 import { deleteApp, initializeApp } from '@firebase/app';
-import { Auth, User } from '../../../src/model/public_types';
```
Comment (renkelvin) [inline packages/auth/test/helpers/integration/helpers.ts:20]: Will webpack 5 raise an error without these changes? If so, I'm curious what does the error look like.

Comment (hsubox76) [reply packages/auth/test/helpers/integration/helpers.ts:20]: There's a type mismatch when `getTestInstance()` (line 37 here) is called by tests such as [anonymous.test.ts](https://github.com/firebase/firebase-js-sdk/blob/c9e2b0b8cd5fd0db3cac7bc3a00629ae34302189/packages/auth/test/integration/flows/anonymous.test.ts), which is using the type from `@firebase/auth`. I think it makes sense to do so as this is an integration test using the built package and this is a helper for integration tests..

Comment (renkelvin) [reply packages/auth/test/helpers/integration/helpers.ts:20]: Yeah, make sense. Thanks for the explanation. I see the design doc now.

**Review by renkelvin — APPROVED** [review]

**Pull request merged**